src/util/codec/bytes.rs

```diff
@@ -112,7 +112,7 @@ pub fn encoded_compact_len(mut encoded: &[u8]) -> usize {
     vn + (encoded.as_ptr() as usize - last_encoded)
 }
 
-pub trait CompactBytesDecoder: NumberDecoder {
+pub trait CompactBytesFromFileDecoder: NumberDecoder {
```

Comment:
Why not remove it?

Reply:
no, since the function `apply_plain_cf_file ` cannot access a slice directly.

Reply:
Why this trait is renamed? From the interface of the trait, I don't see the reason that it is restricted to "from file".